lib/git_worktree.rb

```diff
@@ -6,6 +6,16 @@ class GitWorktree
   DEFAULT_FILE_MODE = 0100644
   LOCK_REFERENCE = 'refs/locks'
 
+  def self.checkout_at(url, directory, options = {})
+    worktree_opts = options.merge(
```

Comment:
do you want `merge` or `reverse_merge`?

Reply:
Both?

Okay, so I am torn, but also in the camp of "doesn't matter" for now since the only two callers of `GitCheckout.checkout_at` are actually internal to this class.

That said, I see why it might make sense to have this be a `.reverse_merge`, but then again, the only options that we don't want overwritten are `:clone` and `:bare` (the other two are set from the arguments), so I think it is fine as is.  Emphasis on _**think**_.